packages/service-library/src/servicelib/redis.py

```diff
@@ -61,35 +88,79 @@ async def ping(self) -> bool:
     async def lock_context(
         self,
         lock_key: str,
-        lock_value: Optional[Union[bytes, str]] = None,
+        lock_value: bytes | str | None = None,
+        *,
+        blocking: bool = False,
+        blocking_timeout_s: NonNegativeFloat = 5,
     ) -> AsyncIterator[Lock]:
-        ttl_lock = None
-        try:
+        """Tries to acquire a lock.
+
+        :param lock_key: unique name of the lock
+        :param lock_value: content of the lock, defaults to None
+        :param blocking: should block here while acquiring the lock, defaults to False
+        :param blocking_timeout_s: time to wait while acquire a lock before giving up, defaults to 5
+
+        :raises CouldNotAcquireLockError: reasons why lock acquisition fails:
+            1. `blocking==False` the lock was already acquired by some other entity
+            2. `blocking==True` timeouts out while waiting for lock to be free (another entity holds the lock)
+        """
+
+        total_lock_duration: datetime.timedelta = _DEFAULT_LOCK_TTL
+        lock_unique_id = f"lock_extender_{uuid4()}"
+
+        ttl_lock: Lock = self._client.lock(
+            name=lock_key,
+            timeout=total_lock_duration.total_seconds(),
+            blocking=blocking,
+            blocking_timeout=blocking_timeout_s,
+        )
 
-            async def _auto_extend_lock(lock: Lock) -> None:
+        if not await ttl_lock.acquire(token=lock_value):
```

Comment:
this might fail right? should this not be in the try...except?

Reply:
If it fails to acquire the lock there is no need to release the lock. It is also correct to raise here if there is an issue. Something else is happening.